Added Workflow.psm1

justsayantan opened the pull request:
Added script to get list of Workflow Items in Tridion Content Manager
By Status or AssignedTo Filter

Commit:
Added Workflow.psm1

Added script to gets a list of Workflow Items in Tridion Content Manager
By Status or AssignedTo Filter

Commit:
Updated the Synopsis

pkjaer committed:
Merge pull request #42 from pkjaer/develop

Develop

pkjaer committed:
Update README.md

pkjaer commented on CoreService/Workflow.psm1:
```diff
@@ -0,0 +1,231 @@
+    For a full list, consult the Content Manager Core Service API Reference Guide documentation 
+    (Tridion.ContentManager.Data.CommunicationManagement.ProcessInstanceData object)
+    .Inputs
+     [string] Status(Not Mandetory): Workflow ProcessInstanceData by status "Active" or "Hstoricle" (Bydefault is will show all the data)
```
Some typos here.

justsayantan replied:
OK. Changes done. Will update in Next Commit

pkjaer commented on CoreService/Workflow.psm1:
```diff
@@ -0,0 +1,231 @@
+		# The TCM URI of the Publication Target to load.
+        [Parameter(ParameterSetName='AssignedToById')]
+		[ValidateNotNullOrEmpty()]
+        [string]$AssignedToById,
```
Maybe we could combine these into one 'AssignedTo' parameter? I think that would be easier to use. 
We could just check if it's a TCM URI (=by ID) or not (=by title).

pkjaer commented on CoreService/Workflow.psm1:
```diff
@@ -0,0 +1,231 @@
+						}					
+					}
+					$filter.ProcessType = [Tridion.ContentManager.CoreService.Client.ProcessType]::Active;
+					$WorkflowItemDetail = $client.GetSystemWideList($filter);
```
In general, we shouldn't use the client directly -- as it cannot be tested with Pester. Use the utility functions instead (which can be mocked up and tested) - such as _GetSystemWideList (in Utilities.ps1)

pkjaer commented on CoreService/Workflow.psm1:
```diff
@@ -0,0 +1,231 @@
+						
+					}
+					Write-Verbose "Loading Publication Target with title '$AssignedToByTitle'..."
+					$filter.ProcessType = [Tridion.ContentManager.CoreService.Client.ProcessType]::Active;
```
Similarly, this cannot be easily tested -- so we should move it to a utility method that can be mocked up.

pkjaer commented on CoreService/Workflow.psm1:
```diff
@@ -0,0 +1,231 @@
+				}
+			    'AssignedToById'
+				{
+					Write-Verbose "Loading Publication Target with Id '$AssignedToById'..."
```
Copy-paste error? :)

justsayantan replied:
OK. Changes done. Will update in Next Commit

pkjaer commented on CoreService/Workflow.psm1:
```diff
@@ -0,0 +1,231 @@
+			    'AssignedToById'
+				{
+					Write-Verbose "Loading Publication Target with Id '$AssignedToById'..."
+					if ($AssignedToById -ne $null)
```
I'm not sure it's worth it to actually load the user/group and validate that it exists etc.
It won't perform very well and in the end if you filter on the wrong ID you should probably just get nothing back.

pkjaer commented on CoreService/Workflow.psm1:
```diff
@@ -0,0 +1,231 @@
+					
+					if($Status -eq $null)
+					{
+						$filter.ProcessType = [Tridion.ContentManager.CoreService.Client.ProcessType]::Any;				
```
Since $Status is already a set with only valid values, you can probably just assign the string directly to $filter.ProcessType (and let PS parse it for you). With the caveat that the empty string option should be replaced with 'Any' instead.

justsayantan replied:
Implemented as per your suggestion.

justsayantan commented on CoreService/Workflow.psm1:
```diff
@@ -0,0 +1,231 @@
+**************************************************
+#>
+
+function Get-TridionWorkflowItems
```
OK. Changes done. Will update in Next Commit

Commit:
Merge remote-tracking branch 'refs/remotes/origin/master' into Script-For-Workflow

Commit:
Change the script as per the Peter's Suggestion

Commit:
Merge remote-tracking branch 'refs/remotes/pkjaer/master' into Script-For-Workflow

pkjaer commented:
I simplified the implementation and added some additional fixes. 
Thanks for your contribution!

justsayantan commented:
Thanks a LOT Peter :)


Regards,
*Sayantan Basu*
Mob: +91 9004079502
Mail:
justsayantan@gmail.com <justsayantan@gmail.com>

On Fri, Jan 5, 2018 at 9:13 PM, Peter Kjaer <notifications@github.com>
wrote:

> I simplified the implementation and added some additional fixes.
> Thanks for your contribution!
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/pkjaer/tridion-powershell-modules/pull/41#issuecomment-355585916>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AK9qXIdofyYjkSaplnur3RSDLkK4W-zhks5tHkMZgaJpZM4RAUyC>
> .
>
